src/main/java/org/kohsuke/github/GHRepository.java

```diff
@@ -843,6 +857,7 @@ public GHPermissionType getPermission(GHUser u) throws IOException {
      * @throws IOException
      *             the io exception
      */
+    @Deprecated
```

Comment:
Now that I'm looking at the api documentation for "[Add user as collaborator](https://developer.github.com/v3/repos/collaborators/#add-user-as-a-collaborator) it doesn't look like permissions are a required in all cases. So, we probably don't want the base methods deprecated. 
```suggestion
```